### PR TITLE
Improve Dockerfile with merged RUNs and a dockerignore whitelist

### PR DESCRIPTION
**Pull request by jakebailey**
This PR overhauls the Dockerfile to reduce the build time and total layers. There are now only two RUNs, before and after the COPY. Only one COPY is made, using .dockerignore to select what should be copied.

On my system:
Before - 36 build stages, 2:15 build time
After - 5 build stages, 1:33 build time

Note that EXPOSE is removed, since it doesn't actually do anything (the `-p` flag does it all).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,45 +1,19 @@
...
-
-COPY docker/init.sh /etc/init.sh
-RUN chmod +x /etc/init.sh
-
-COPY exampleCourse /exampleCourse
-RUN mkdir /course
-COPY docker/config.json /prairielearn
+RUN yum -y install \
+    https://download.postgresql.org/pub/repos/yum/9.6/redhat/rhel-7-x86_64/pgdg-centos96-9.6-3.noarch.rpm\
+    https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm \
```
Comment (lotrfan) [inline Dockerfile]: `epel-release` should work instead of a URL here

Comment (lotrfan) [inline Dockerfile]: You can also do a `yum clean all` to remove the caches, which should make the image a bit smaller.

Comment (jakebailey) [inline Dockerfile]: Yep, I was already doing that last one when you commented.

Comment (jakebailey) [inline Dockerfile]: Done. Very good size reduction.

```
REPOSITORY                  TAG                 IMAGE ID            CREATED             SIZE
pl                          latest              f8f54ef9f3be        35 seconds ago      429.1 MB
prairielearn/prairielearn   latest              ad67adc84a5e        About an hour ago   562.9 MB
centos                      7                   67591570dd29        4 weeks ago         191.8 MB
```

**Pull request merged**